### PR TITLE
Created AirQuality App and Displayed the data using Openweather API

### PR DESCRIPTION
**Pull request by ginabeki**
In this milestone, I did the following;
- Use React components to structure the project.
- Used React props.
- Used React Router.
- Connected React and Redux.
- Handled events in the app.
- Wrote integration tests with a React testing library.
- Used styles in a React app.
- Used React life cycle methods.
- Applied React best practices and language style guides in code.
- Used store, actions, and reducers in the app

Here's the [link](https://www.loom.com/share/9a544a1157c747b09767cc708ee44e55) to my presentation. I had to remove it from my readme since it kept on throwing the gitguardian error.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,7 @@
...
 
 <div align="center">
- 
-  <img src="murple_logo.png" alt="logo" width="140"  height="auto" />
-  <br/>
-
-  <h3><b>Microverse README Template</b></h3>
+   <h3><b>Microverse README Template</b></h3>
 
 </div>
```
Comment (FrederickMih) [inline README.md:6]: - [x] Let's kindly give appropriate credit to Nelson Sakwa, the author of the original design in the "Acknowledgements" section of the README.md file.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand:  @ginabeki, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!




Screenshot from 2022-10-20 09-38-51

## Highlights :green_circle: 

- One color was consistent throughout the app :heavy_check_mark: 
- Redux was used for state management :+1: 
- PR is professional :+1: 
- Unit tests are passing without errors :+1: 



## Suggested changes

**NB**. Since fewer files were changed in this project, I'm going to leave comments here:

- [x] Let's kindly make sure that the code structure is organized in a logical manner. The components folder should contain only your app's components. redux, pages, _tests_, etc folders should be found in the src directory as seen in the screenshot below:

Screenshot from 2023-01-06 13-18-17
Proper project structure makes it easier to develop and maintain the application over time.


- [x] All the list of categories(cities) should be retrieved from the API and displayed on the home page when the home page load before filtering. On your home page, only six items are displayed which does not concretely define the word **list** as seen in the requirements. This list should be difficult to browse, that's where the search bar becomes very important. Also, note that filtering should be done on its own on a stateless **Filter** component.
- [x] Let's kindly fix all the bugs :bug: and stylistic errors flagged by Linters. Linter is a great tool that helps to ensure that your code does not include any structural issues which can make your code harder to maintain in the future. Let's not disable them. Kindly walk through all the components you disabled linters and fix :wrench:  the various issues
- [x] Hi @ginabeki, kindly note that your app needs to be working properly(Both local and remote). I got the error on the screenshot below when clicking on any city. Kindly fix the error :bug: 
Screenshot from 2023-01-06 13-44-18

- [ ] Based on the requirements of this project, we are supposed to follow the design guidelines including Typography(font face, size, and weight) and layout(composition and space between elements). Let's kindly check the difference in the compared screenshot below and make the necessary changes.

| Yours | Design guidelines |
| --- | --- |
| Screenshot from 2023-01-06 13-50-21 | Screenshot from 2022-12-02 19-27-09 |

NB. Having a nice design increases the chances of a job as recruiters will be checking projects on GitHub.

- [x] **Testing:** Let's kindly add more tests for the app. Pure functions in Redux code should be tested. App has integration tests using React Testing Library.









_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by victorokoroji — CHANGES_REQUESTED** [review]
Hi @ginabeki, 

Great work on making the changes requested by a previous reviewer 👏 🔥 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



## Top highlight 🎯

- The PR title and description are professional and descriptive ✅ 
- The commit messages are meaningful ✅ 
- Linters are working properly ✅ 


 ## Required Changes ♻️

- [ ] As the previous reviewer has suggested, i cant seem to get the details of each location data on my local machine.
I got the same error on the screenshot below when clicking on any city. Kindly fix the error 🐛

image

I think you can explore other ways of displaying the details such as useParams, useLocation etc. Just ensure the data is fetched locally as well as it was correctly fetched on the live site 👍🏽 

- [ ] As the previous reviewer has earlier suggested, on your home page, only six items are displayed which does not concretely define the word list as seen in the requirements. 
 
I can see that you manually added the six locations in your api.js file. I think you can get a free API that will get you the longitude and latitude of all locations you are targeting. Displaying six data is rather too small and it should be gotten from a real API and not yu manually adding them 👍🏽  
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rloterh — COMMENTED** [review]
Hi Gina,
 
 While you made a great effort in this project, unfortunately, I cannot proceed to review your code.
 
 ## Invalid Code Review Request
Kindly ensure that the last commit have a green (✅) checkmark!
 
Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
 
 Cheers and Happy coding!👏👏👏
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

@rloterh Hi, I have updated the api key inside an environment variable, but I still get the git guardian error, is there a way you can help me solve this?

**Comment by ginabeki** [comment]
- Here are the env keys to run the project on your local
REACT_APP_URL = 'http://api.openweathermap.org/data/2.5/air_pollution/forecast'
REACT_APP_KEY = '&appid=c5e60d49c527e846f710b5b769e46a92'

File: README.md [hunk 2]
```diff
@@ -1,11 +1,7 @@
 <a name="readme-top"></a>
 
 <div align="center">
```
Comment (MeqdamAlqudah) [inline README.md:3]: - [x] @ginabeki Please just add a .env file to your root directory and then you can use them like this 
image

and it is working 
image

- [ Optional] change the color of the back button on the detail page
image

**Review by MeqdamAlqudah — CHANGES_REQUESTED** [review]
Hi @ginabeki , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Highlights
:+1: Readme file
:+1: using  Redux 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 2 above)
Comment (ginabeki) [reply README.md:3]: Hi @MeqdamAlqudah , I have implemented the suggested changes, kindly review the updated version

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand: @ginabeki, 

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

## Highlights :green_circle: 

- Professional PR :heavy_check_mark: 
- All Linter checks are passing :heavy_check_mark: 

Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**